src/main/java/ruiz/Parser.java

```diff
@@ -0,0 +1,159 @@
+     * @return description of the ToDo input by the user.
+     * @throws BotException if the ToDo is in an invalid format.
+     */
+    public String getTodoDescription(String input) throws BotException {
+        if (input.split(" ", 2).length <= 1) {
+            throw new BotException("OOPS!!! The description of a todo cannot be empty.");
+        }
+        return input.split(" ", 2)[1];
+    }
+
+    /**
+     * Returns the description of the Deadline input by the user.
+     *
+     * @param input input of the user
+     * @return description of the deadline input by the user.
+     * @throws BotException if the deadline is in an invalid format.
+     */
+    public String getDeadlineDescription(String input) throws BotException {
+        if (input.split("/by", 2).length <= 1) {
+            throw new BotException("OOPS!!! The description the deadline is incomplete,"
+                    + " it should be in the format 'deadline *** /by ***'");
+        }
+        return input.split(" ", 2)[1].split(" /by ")[0];
+    }
+
+    /**
+     * Returns the time the deadline must be completed by input by the user.
+     *
+     * @param input input of the user
+     * @return time the deadline must be completed by input by the user.
+     * @throws BotException if the deadline is in an invalid format.
+     */
+    public String getBy(String input) throws BotException {
+        if (input.split("/by", 2).length <= 1) {
+            throw new BotException("OOPS!!! The description the deadline is incomplete,"
+                    + " it should be in the 'format deadline *** /by ***'"
+                    + "");
+        }
+        return input.split(" ", 2)[1].split(" /by ")[1];
+    }
+
+    /**
+     * Returns the description of the Event input by the user.
+     *
+     * @param input input of the user
+     * @return description of the Event input by the user.
+     * @throws BotException if the Event is in an invalid format.
+     */
+    public String getEventDescription(String input) throws BotException {
+        if (input.split("/").length <= 2) {
+            throw new BotException("OOPS!!! The description the event is incomplete."
+                    + "it should be in the format 'event *** /from *** /to ***'");
+        }
+        return input.split(" ", 2)[1].split(" /from")[0];
+    }
+
+    /**
+     * Returns the time the event begins by input by the user.
+     *
+     * @param input input of the user
+     * @return time the event begins by input by the user.
+     * @throws BotException if the event is in an invalid format.
+     */
+    public String getFrom(String input) throws BotException {
+        if (input.split("/").length <= 2) {
+            throw new BotException("OOPS!!! The description the event is incomplete."
+                    + "it should be in the format 'event *** /from *** /to ***'");
+        }
+        return input.split(" ", 2)[1].split(" /from ")[1].split(" /")[0];
```

Comment:
Its not clear immediately what the chain of splits is supposed to achieve to me. Maybe each split can be split off into its own descriptive function or something similar to improve readability?